### ISSUE TITLE
Cannot build with WixToolset.DifxApp.wixext in a Library .wixproj

### ISSUE THREAD

**Issue by avivanoff** [wixtoolset/issues#7625]
Re-opening #7558 as it was closed and I do believe there is a bug. These are my findings.

If you build the [original sample project](https://github.com/wixtoolset/issues/files/11785157/Library1.zip), you get the following error:

```
1>D:\a\wix\wix\src\ext\DifxApp\wixlib\DifxAppExtension.wxs(11): error WIX0103: Cannot find the Binary file 'wix-ir/DIFxApp.dll'. The following paths were checked: wix-ir/DIFxApp.dll
1>D:\a\wix\wix\src\ext\DifxApp\wixlib\DifxAppExtension.wxs(12): error WIX0103: Cannot find the Binary file 'wix-ir/DIFxAppA.dll'. The following paths were checked: wix-ir/DIFxAppA.dll
```

According to @barnson this is expected as you are expected add a reference to an external .wixlib. After examining WixToolset.DifxApp.wixext.targets from the referenced wixtoolset.difxapp.wixext 4.0.1 NuGet package, you can see that the proper external wixlib is being referenced:

`<WixLibrary Include="$(MSBuildThisFileDirectory)..\$(WixToolsetExtensionPackageFolder)\difxapp_$(InstallerPlatform).wixlib" />
`

Also, in Visual Studio Solution Explorer, you can clearly see that the library is indeed being referenced:

<img width="227" alt="Screenshot 2023-07-18 at 11 59 30" src="https://github.com/wixtoolset/issues/assets/29612120/df36225b-b94b-478d-8ffd-f5f3fcaae673">

More that that, if you set `BindFiles` to `false` in the sample project, it builds successfully and `DIFxApp.dll` and `DIFxAppA.dll` are being embedded in `Library1.wixlib` under `wix-ir` folder.

Am I missing something?

**Comment by oleg-shilo** [comment]
I do not know if it helps you before it is addressed by the WiX team but you can have a look at the attached sample it builds the msi just fine.

This is the [script/project location](https://github.com/oleg-shilo/wixsharp/blob/wix-v4-master/Source/src/WixSharp.Samples/Wix%23%20Samples/Driver/setup.cs)
```C#
static void Main()
{
    var project = new Project("MyProduct",
                      new Dir(@"%ProgramFiles%\My Company\My Device",
                          new File("driver.sys",
                              new DriverInstaller
                              {
                                  AddRemovePrograms = false,
                                  DeleteFiles = false,
                                  Legacy = true,
                                  PlugAndPlayPrompt = false,
                                  Sequence = 1,
                                  Architecture = DriverArchitecture.x64
                              })));

    project.GUID = new Guid("6f330b47-2577-43ad-9095-1861ba25889b");

    project.LibFiles.Add(@"%userprofile%\.wix\extensions\WixToolset.DifxApp.wixext\4.0.0\wixext4\difxapp_x64.wixlib");
 
    project.PreserveTempFiles = true;

    project.BuildMsi();
}
```
 
[Driver_working sample.zip](https://github.com/wixtoolset/issues/files/12089885/Driver_working.sample.zip)

**Comment by robmen** [comment]
The root issue here is that Library projects do not support binding files from .wixlibs in WiX v4 (maybe this worked in v3, I don't remember). The `WixToolset.DifxApp.wixext` is including a standalone .wixlib using the `WixLibrary` element which creates exactly this situation.

If you make the consuming project anything but a Library project, the build will work.

tl;dr:  `WixToolset.DifxApp.wixext` cannot be referenced by a Library project at this time.

**Comment by robmen** [comment]
Though of a quick workaround that you can add to your Library1.wixproj:

```xml
<Target Name="RemoveWixLibraries" BeforeTargets="PrepareForBuild">
  <ItemGroup>
    <WixLibrary Remove="@(WixLibrary)" />
  </ItemGroup>
</Target>
```

**Comment by oleg-shilo** [comment]
Rob we had this topic [discussed](https://github.com/orgs/wixtoolset/discussions/7571) with @BMurri. I left that topic assuming that his suggestion to "add the .wixlib" was an adequate solution for the problem. So I shared it here as well, just two posts above. 

Though you downvoted my post. This makes me think that, in your opinion, the solution based on BMurri's hint was not in fact adequate and should not be used. 

Did I understand you correctly?

**Comment by oleg-shilo** [comment]
Don't get me wrong, Which of the discussions is accurate and which is wrong (issues/7625 vs discussions/7571)
I am only interested in knowing if WiX4 can be used now to implement `Driver` as it is described in the documentation.

**Comment by oleg-shilo** [comment]
OK, I see.
I was bringing that syntax the same way as people do with pseudo code when they are discussing programming challenges on forums. But the actual solution (the attachment) I shared with @avivanoff is a pure WiX4 syntax.

But since it was not clear, I have edited the post and made an extra emphasis on that.

**Comment by barnson** [comment]
Automatically adding the Difx .wixlib is causing the problem. Given that that is rarely used and Difx is deprecated and will be removed in WiX v5, we're leaving this alone.
